astropy/io/misc/asdf/tags/transform/properties.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+
+from astropy.modeling.bounding_box import ModelBoundingBox, CompoundBoundingBox
+from astropy.io.misc.asdf.types import AstropyAsdfType
+
+
+__all__ = ["ModelBoundingBoxType", "CompoundBoundingBoxType"]
+
+
+class ModelBoundingBoxType(AstropyAsdfType):
+    name = 'transform/property/bounding_box'
+    version = '1.0.0'
+    types = ['astropy.modeling.bounding_box.ModelBoundingBox']
+
+    @classmethod
+    def to_tree(cls, bbox, cts):
+        if isinstance(bbox, ModelBoundingBox):
+            return {
+                'intervals': {
+                    _input: list(interval) for _input, interval in bbox.intervals.items()
+                },
+                'ignore': list(bbox.ignored),
+                'order': bbox.order
+            }
+        else:
+            raise TypeError(f"{bbox} is not a valid ModelBoundingBox!")
```

Comment:
This doesn't hurt anything, but I don't usually bother to test the type of the input object since the framework is supposed to guarantee that it has the expected type.